docs/pages/hdmi_cec_source_tests.md

```diff
@@ -0,0 +1,176 @@
+# HDMI CEC Test Document
+
+## Version History
+
+| Date(DD/MM/YY) | Comment | Version |
+| -------------- | ------- | ------- |
+| 13/02/24 | First Release | 1.0.0 |
+
+## Table of Contents
+
+- [Acronyms, Terms and Abbreviations](#acronyms-terms-and-abbreviations)
+- [Overview](#overview)
+- [HDMI-CEC RDK HAL Functionality](#hdmi-cec-rdk-hal-functionality)
+
+## Acronyms, Terms and Abbreviations
+
+- `CEC` - Consumer Electronics Control
+- `HAL` - Hardware Abstraction layer
+- `HDMI` - High Definition Multimedia Interface
+- `L2` - Level 2 Testing ()
+- `L3` - Level 3 Testing ()
+- `Sink device` - Any piece of equipment or technology that provides an input signal or data to another device or system.
+- `Source Device` - Any piece of equipment or technology that receives an input signal or data from another device or system.
+
+## Scope
+
+This document defines the requirements for testing the HDMI CEC device from a level 2/3 based on being a source device.
+
+## Overview
+
+Consumer Electronics Control (CEC) is a single-wire bidirectional bus within an HDMI system, facilitating communication among interconnected products. HDMI-CEC establishes a protocol enabling high-level control functions between audiovisual devices linked via an HDMI network, facilitating communication and control among them. Communication can occur in either Direct messaging mode or Broadcast mode. 
+
+## HDMI-CEC RDK HAL Functionality
+
+The HAL layers within RDK serve as a bridge between the underlying low-level SoC drivers and the higher-level RDK layers that utilize the functionality offered by these HAL functions. Specifically concerning the CEC Module, the HAL layers facilitate the following functionalities:
+
+- Get Logical address
+- Get Physical address
+- Syncronous transmission, and communicating via hotplug connectivity
+
+## Test scenarios
+
+The HAL CEC layer facilitates the transmission and reception of CEC information on the CEC bus. It does not handle any specific opcode commands, nor does it validate supported HAL CEC opcodes for sending or receiving.
+
+It is the responsibility of the caller to manage the opcodes. The current test cases will verify responses from connected devices for a subset of opcodes as part of the testing process.
+
+|S.No.|Test Functionality|Description|
+|-----|------------------|-----------|
+| 1 |[Logical address](#logical-address-discovery)|Facilitating the Discovery of logical addresses getting, and removing the logical address of the device (for sink devices) |
```

Comment:
let's keep just with source devices. don't include the sink devices.  Also on source devices, no logical address removal.  Its only discovery in open and getlogical address.

Reply:
Updated as requested